atuin-client/src/import/xonsh.rs

```diff
@@ -0,0 +1,238 @@
+use std::env;
+use std::fs::{self, File};
+use std::path::{Path, PathBuf};
+
+use async_trait::async_trait;
+use directories::BaseDirs;
+use eyre::{eyre, Result};
+use serde::Deserialize;
+use time::OffsetDateTime;
+use uuid::timestamp::{context::NoContext, Timestamp};
+use uuid::Uuid;
+
+use super::{Importer, Loader};
+use crate::history::History;
+
+// Note: both HistoryFile and HistoryData have other keys present in the JSON, we don't
```

Comment:
Most of it is just metadata that Xonsh uses to lazily access the JSON so that it doesn't have to read/write the whole file on every update, they talk about it a bit [here](https://xon.sh/tutorial_hist.html#exciting-technical-detail-lazy-json).

The only other things I'm ignoring are the env (as you mentioned below) and the session timestamp. Arguably I should be using the session timestamp to generate the session UUIDv7 (which replaces the existing UUIDv4 session id), but I think I didn't realize it existed when I was implementing that so I just used the timestamp of the first command. I didn't think this was worth going back and fixing, but I'd be happy to if you think it would be better. :) 

I can mention in the docs that the import ignores env, since that might matter to people. Does the rest of it merit a mention in the docs, or should I just make the comment here a little more detailed?

Reply:
Sorry I totally missed the follow up here, only noticed when going back through.

Thanks for the explainer! Xonsh has some interesting history stuff going on for sure. I'm happy with the docs as they are now :)